ibm/service/secretsmanager/resource_ibm_sm_public_certificate.go

```diff
@@ -887,3 +930,222 @@ func ConcreteListToStringSlice(elements []interface{}, commonName string) ([]str
 
 	return output, nil
 }
+
+func setChallengesWithAkamaiAndValidateManualDns(context context.Context, d *schema.ResourceData, meta interface{}, secret *secretsmanagerv2.PublicCertificate, secretsManagerClient *secretsmanagerv2.SecretsManagerV2) diag.Diagnostics {
+	edgerc := d.Get("akamai").([]interface{})[0].(map[string]interface{})["edgerc"].(string)
+	configSection := d.Get("akamai").([]interface{})[0].(map[string]interface{})["config_section"].(string)
+	config, err := edgegrid.InitEdgeRc(edgerc, configSection)
+	if err != nil {
+		resourceIbmSmPublicCertificateDelete(context, d, meta)
```

Comment:
Why in case of error  we are deleting the instance?
This will remvoe the instance from backend

Reply:
@hkantare because It's a cleaning before returning the error. The "set challenges" operation is separate to the secret creation, so if a secret has been created but "set challenges" failed, the user should get an error and the secret should not be created (and that's why it's being deleted).

Reply:
@hkantare This section is only used when creating a new secret, and this is a new feature - old configurations didn't use this specific type of secret that use this code. So the scenario you described is not possible.